libs/core/threading_base/include/hpx/threading_base/scheduler_base.hpp

```diff
@@ -369,6 +369,12 @@ namespace hpx { namespace threads { namespace policies {
             return work_count;
         }
 
+        // almost all schedulers support direct execution
+        virtual bool supports_direct_execution() const
+        {
+            return true;
```

Comment:
Would it be better to make this opt-in?

Reply:
I didn't think of this option. This is mostly a mechanism to disable direct execution on the shared_priority_scheduler.

Reply:
I think it's the only correct default since a scheduler has to explicitly support direct execution. Or is there some other reason for the shared priority scheduler not working with direct execution?

Reply:
The shared priority scheduler relies on exactly placing threads onto particular queues (i.e. cores). Direct execution might interfere, so I decided not to support it on that scheduler.

Reply:
Relies? Or supports? The default scheduler also allows placing threads on particular queues (worker threads), so that alone can't be the reason for not supporting direct execution.

Reply:
Well, the whole point of this scheduler is to try as hard as possible to run threads on the core as specified, would you agree? Other schedulers are quite loose about the hint.

Reply:
I'm not sure it's useful to start quantifying how hard a scheduler tries to run a thread on a specified core. The default scheduler tries pretty hard as well. It schedules things on a core given by the hint if there is a hint. It just doesn't know what to do with numa hints.

It sounds like `runs_as_child = true` should only be allowed if the hint mode is `thread_schedule_hint_mode::none`, independently of the scheduler. Do you think that sounds reasonable?

Reply:
Frankly, there was more to it than just the thread placement, I just can't recall what it was. Let me get back to it and figure out why I thought that the only way was to disable direct execution.